include/fastdds/rtps/builtin/data/PublicationBuiltinTopicData.hpp

```diff
@@ -0,0 +1,138 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file PublicationBuiltinTopicData.hpp
+ */
+
+#ifndef FASTDDS_RTPS_BUILTIN_DATA__PUBLICATIONBUILTINTOPICDATA_HPP
+#define FASTDDS_RTPS_BUILTIN_DATA__PUBLICATIONBUILTINTOPICDATA_HPP
+
+#include <cstdint>
+#include <string>
+
+#include <fastdds/dds/core/policy/QosPolicies.hpp>
+#include <fastdds/rtps/builtin/data/BuiltinTopicKey.hpp>
+#include <fastdds/rtps/common/Guid.hpp>
+#include <fastdds/rtps/common/RemoteLocators.hpp>
+#include <fastdds/rtps/common/Types.hpp>
+
+namespace eprosima {
+namespace fastdds {
+namespace rtps {
+
+struct PublicationBuiltinTopicData
+{
+    /// Builtin topic Key
+    BuiltinTopicKey_t key;
+
+    /// Builtin participant topic Key
+    BuiltinTopicKey_t participant_key;
+
+    /// Topic name
+    std::string topic_name;
```

Comment:
I guess that making these fields as `fastcdr::string_255` makes no difference

Reply:
I think it makes sense to change it.